__tests__/app.test.js

```diff
@@ -158,5 +158,135 @@ describe('app.js tests', () => {
+          expect(Array.isArray(body)).toBe(false);
+          expect(Array.isArray(body.updated)).toBe(false);
+          expect(typeof body).toBe('object');
+          expect(typeof body.updated).toBe('object');
+        });
+    });
+    test('object with positive value increases', () => {
+      return request(app)
+        .patch('/api/reviews/1')
+        .send({
+          inc_votes: 10,
+        })
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.updated.votes).toBe(11);
+        });
+    });
+    test('object with negative value decreases', () => {
+      return request(app)
+        .patch('/api/reviews/1')
+        .send({
+          inc_votes: -1,
+        })
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.updated.votes).toBe(0);
+        });
+    });
+    test('object with negative value cant go below 0', () => {
+      return request(app)
+        .patch('/api/reviews/1')
+        .send({
+          inc_votes: -10,
+        })
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.updated.votes).toBe(0);
+        });
+    });
+    test('it returns object of updated review', () => {
+      return request(app)
+        .patch('/api/reviews/1')
+        .send({
+          inc_votes: 10,
+        })
+        .expect(200)
+        .then(({ body }) => {
+          const obj_1 = {
+            review_id: 1,
+            title: 'Agricola',
+            designer: 'Uwe Rosenberg',
+            owner: 'mallionaire',
+            review_img_url:
+              'https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png',
+            review_body: 'Farmyard fun!',
+            category: 'euro game',
+            created_at: '2021-01-18T10:00:20.514Z',
+            votes: 11,
+          };
+          expect(body.updated).toEqual(obj_1);
+        });
+    });
+
+    // error handling
+    test('400: missing required fields for patch input', () => {
+      return request(app)
+        .patch('/api/reviews/1')
+        .send({})
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toEqual('Invalid input, missing key or value');
+        });
+    });
+    test('400: incorrect value type for patch input ', () => {
+      return request(app)
+        .patch('/api/reviews/1')
+        .send({
+          inc_votes: 'cow',
+        })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toEqual('Invalid input syntax');
+        });
+    });
+    test('400: incorrect value type for patch input ', () => {
+      return request(app)
+        .patch('/api/reviews/1')
+        .send({
+          inc_votes: 5.39,
+        })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toEqual('Invalid input syntax');
+        });
+    });
+    test('400: Id is too large', () => {
+      return request(app)
+        .patch('/api/reviews/1010101010101')
+        .send({
```

Comment:
I don't think you need this test. This should fall under 404 as you could have a review with id 1010101010101 in the future!

Reply:
number is too large for INT datatype in SQL so throws a 400 error but I wouldn't put it in the same class as "invalid syntax" as it IS a number still

Reply:
Ahh that makes sense! Ignore me then!